test/Chainweb/Test/Pact/PactInProcApi.hs

```diff
@@ -412,6 +413,179 @@ chainweb213Test bdb mpRefIO pact = do
         ]
     runCut' = runCut testVersion bdb pact (offsetBlockTime second) zeroNoncer noMiner
 
+pact43UpgradeTest :: TestBlockDb -> IO (IORef MemPoolAccess) -> WebPactExecutionService -> IO ()
+pact43UpgradeTest bdb mpRefIO pact = do
+
+  -- run past genesis, upgrades
+  forM_ [(1::Int)..27] $ \_i -> runCut'
+
+  -- run block 28, pre fork
+  setOneShotMempool mpRefIO getBlock1
+  runCut'
+  pwo28 <- getPWO bdb cid
+  tx28_0 <- txResult "pwo28" 0 pwo28
+  assertEqual "Old gas cost" 120332 (_crGas tx28_0)
+
+  -- run block 29, pre fork
+  setOneShotMempool mpRefIO getBlock2
+  runCut'
+  pwo29 <- getPWO bdb cid
+  tx29_0 <- txResult "pwo29" 0 pwo29
+  assertEqual
+    "Should not resolve new pact native: continue"
+    (Just "Cannot resolve \"continue\"")
+    (tx29_0 ^? crResult . to _pactResult . _Left . to peDoc)
+
+  -- run block 30, pre-fork
+  setOneShotMempool mpRefIO getBlock3
+  runCut'
+  pwo30 <- getPWO bdb cid
+
+  tx30_0 <- txResult "pwo30" 0 pwo30
+  assertEqual
+    "Should not resolve new pact native: create-principal"
+    (Just "Cannot resolve create-principal")
+    (tx30_0 ^? crResult . to _pactResult . _Left . to peDoc)
+
+  tx30_1 <- txResult "pwo30" 1 pwo30
+  assertEqual
+    "Should not resolve new pact natives: validate-principal"
+    (Just "Cannot resolve validate-principal")
+    (tx30_1 ^? crResult . to _pactResult . _Left . to peDoc)
+
+  -- run block 31, post-fork
+  setOneShotMempool mpRefIO postForkBlock1
```

Comment:
```suggestion
  setOneShotMempool mpRefIO block31
```

Also, the following _really_ should be in one block. The previous ones didn't slow the test because you had to get to block 30, but doing extra blocks post-fork makes the test slower.

Reply:
Fixed.